llvm/lib/Target/WebAssembly/WebAssemblyExplicitLocals.cpp

```diff
@@ -379,6 +385,13 @@ bool WebAssemblyExplicitLocals::runOnMachineFunction(MachineFunction &MF) {
     Changed = true;
   }
 
+  {
+    auto RL = Reg2Local.find(MFI.SPVReg);
+    if (RL != Reg2Local.end()) {
+      MFI.SPLocal = RL->second;
+    }
+  }
```

Comment:
This search either doesn't run, doesn't run early enough, or doesn't succeed.

I'm not sure what the reason is, but the effect is that `SPLocal` can remain uninitialized, which produces non-determinism in two observable ways:
1. the value written to `.debug_info` can change between runs (and it's pretty large, suggesting it's not actually valid)
2. the condition of `if (FBL.TI.Offset >= 0) {` in `DwarfCompileUnit.cpp`, which gates the emission of `DW_AT_frame_base`, can change between runs

I've confirmed  this by initializing `SPLocal` to `42` in `WebAssemblyFunctionInfo`'s constructor, which makes:
```sh
llvm-dwarfdump -debug-info foo.wasm | grep DW_AT_frame_base
```
output *only* lines like this:
```c
DW_AT_frame_base (DW_OP_WASM_location 0x0 +42, DW_OP_stack_value)
```

Reply:
@eddyb you are right. I think was addressed by its new version at https://reviews.llvm.org/D69807 -- it will update it soon at rust side (hopefully with LLVM landing too 🤞)

Reply:
Patching using of the above might take time, I submitted https://github.com/rust-lang/llvm-project/pull/27